soccer/MainWindow.cpp

```diff
@@ -267,52 +262,67 @@ void MainWindow::updateViews() {
 
         _logMemory->setText(
             QString("Log: %1/%2 %3 kiB")
-                .arg(QString::number(_processor->logger().numFrames()),
-                     QString::number(_processor->logger().maxFrames()),
+                .arg(QString::number(_processor->logger().size()),
+                     QString::number(_processor->logger().capacity()),
                      QString::number((_processor->logger().spaceUsed() + 512) /
                                      1024)));
     }
 
+    auto value = _ui.logHistoryLocation->value();
+    if (static_cast<int>(value) != frameNumber()) {
+        //_doubleFrameNumber = value;
+        //_playbackRate = 0;
+    }
+    //    emit historyLocationChanged(_doubleFrameNumber);
+
     // Advance log history
-    int liveFrameNumber = _processor->logger().lastFrameNumber();
-    if (_live) {
+    int liveFrameNumber = _processor->logger().currentFrameNumber();
+    if (live()) {
         _doubleFrameNumber = liveFrameNumber;
     } else {
-        _doubleFrameNumber += _playbackRate;
+        _doubleFrameNumber += *_playbackRate;
 
-        int minFrame = _processor->logger().firstFrameNumber();
-        int maxFrame = _processor->logger().lastFrameNumber();
+        int minFrame = _processor->logger().firstFrameNumber() + 10;
```

Comment:
Oh it just gives the processor some buffer to work with since the buffer will keep moving as we display them.